examples/cuda/examples/maxwell/snr.cuh

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) NVIDIA
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "common.cuh"
+
+#include <schedulers/stream.cuh>
+#include <schedulers/inline_scheduler.hpp>
+#include <schedulers/static_thread_pool.hpp>
+
+namespace ex = std::execution;
+namespace stream = example::cuda::stream;
+
+using example::cuda::is_on_gpu;
+
+namespace repeat_n_detail {
+
+  struct sink_receiver : stream::receiver_base_t {
+    friend void tag_invoke(ex::set_value_t, sink_receiver &&, auto&&...) noexcept {}
+    friend void tag_invoke(ex::set_error_t, sink_receiver &&, auto&&) noexcept {}
+    friend void tag_invoke(ex::set_stopped_t, sink_receiver &&) noexcept {}
+    friend ex::__empty_env tag_invoke(ex::get_env_t, sink_receiver) noexcept {
+      return {};
+    }
+  };
+
+  template <class SenderId, class ReceiverId>
+    struct operation_state_t {
+      using Sender = std::__t<SenderId>;
```

Comment:
Given that ADL-isolation is an important tool for S&R, we should provide some standard facility that is not __ugly

When we were trying to get another reference implementation going I used this hacked up thingy:

https://github.com/dietmarkuehl/P2300/blob/main/include/p2300/non_associated.hpp

No change requested, just food for thought

Reply:
You mean, standardize it as part of P2300?

Reply:
Yes, a user should be able to write senders without having to use implementation reserved keywords or rebuild standard facilities by hand.

Also it would help to actually constraint based on a given standard facility so that it is less often forgotten